dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -473,6 +475,42 @@ public String toString() {
         public void close() {}
     }
 
+    /**
+     * {@link ModulatingOutputStream} limits the size of individual writes to the wrapped {@link OutputStream}
+     * in order to prevent degraded performance on large buffers as described in
+     * <a href="https://github.com/palantir/hadoop-crypto/pull/586">hadoop-crypto#586</a>.
+     */
+    static final class ModulatingOutputStream extends FilterOutputStream {
```

Comment:
Adjusted naming from `ChunkingOutputStream` to `ModulatingOutputStream` to avoid confusion with multipart chunks since this is just wrapping & writing to a single OutputStream in 16KB blocks